### PR TITLE
Refactors admin PMs to be hopefully more resilient

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

MSO got me curious, so I refactored admin help code doin my best to follow https://github.com/tgstation/tgstation/pull/68324#issuecomment-1179773970
The basic idea is try and make behavior as clear as possible, and keep data extraction separate from logic
Also fail loud, in this case meaning any operation that would cause the pm to fail to send should send an error message to the sender.

I'm splitting the main proc up into subsections, in an attempt to make it easier to reason about and work with.
As a side effect of this, some operations need to be done more then once to avoid potential debugger escalation.

Related to this, I'm also no longer directly passing around client references through the procs. I instead pass the result of a pretty basic proc into each, to avoid issues with sleeping and such.

Changes the backup for replying to someone who logs out, instead of using the first person to send something to the ticket as our backup, we use the recipient's ckey.

Converts all uses of the linkify span to a macro. One of these doesn't have a closing span, not sure if that's a bug or a feature

Also I just cleaned things up, because the old state of things really bothered me. Too hard to read/understand meaning, and too verbose.
I've also fixed some logic errors, like being unable to respond to tgs ahelps, or a few runtimes in tgs handling.
Did you know secret ckeys, despite claiming to be unique, weren't? 
The hope is this will prevent failure in the tickets system in future

## Why It's Good For The Game

I really don't want to go back to walking up and down stairs trying to get the sand bug to trigger, guess it starts tomorrow.

## Changelog

:cl:
admin: Wooooo spoooky ADMIN HELP BACKEND HAS CHANGED. If you run into any strange behavior or dropped messages, REPORT THEM
/:cl:

### CODE REVIEW

File: code/__DEFINES/span.dm [hunk 1]
```diff
@@ -60,6 +60,7 @@
 #define span_info(str) ("<span class='info'>" + str + "</span>")
 #define span_infoplain(str) ("<span class='infoplain'>" + str + "</span>")
 #define span_interface(str) ("<span class='interface'>" + str + "</span>")
+#define span_linkfiy(str) ("<span class='linkify'>" + str + "</span>")
```
Comment (TrashDoxx) [inline code/__DEFINES/span.dm:63]: ehh.. linkfiy? isn't that a mistake? here and further in the code.

Comment (TrashDoxx) [reply code/__DEFINES/span.dm]: me too

File: code/modules/admin/verbs/adminpm.dm [hunk 2]
```diff
@@ -113,333 +173,591 @@
...
+	if(IsAdminAdvancedProcCall())
+		return
+
+	// First we filter, because these procs can be called by anyone with debug, and I don't trust that check
+	// gotta make sure none's fucking about
+	log_message = adminpm_filter_text(ambiguious_recipient, log_message)
+	if(!log_message)
+		return
+
+	var/raw_messsage = log_message
```
Comment (san7890) [inline code/modules/admin/verbs/adminpm.dm:517]: Typo? Here and subsequent lines.

**Comment by jlsnow301** [comment]
i love that you've used _lad_ and _boi_ as mob var names here, getting tired of thinking up crafty names myself

**Review by MrsTonedOne — APPROVED** [review]
Thanks for the insanely fast turn around on this.

File: code/modules/admin/verbs/adminpm.dm [hunk 3]
```diff
@@ -25,80 +42,123 @@
...
+	var/request = "Private message to"
+	if(recipient_fake_key)
+		request = "[request] an Administrator."
+	else
+		request = "[request] [formatted_recipient_ckey]."
 
-	if(AH?.opening_responders && length(AH.ticket_interactions) == 1)
-		SEND_SOUND(src, sound('sound/machines/buzz-sigh.ogg', volume=30))
-		message_prompt += "\n\n**This ticket is already being responded to by: [english_list(AH.opening_responders)]**"
+	var/messsage = input(src, message_prompt, request) as message|null
```
Comment (san7890) [inline code/modules/admin/verbs/adminpm.dm:132]: even more messsage spellcheck here and subsequent lines

**Comment by LemonInTheDark** [comment]
> Thanks for the insanely fast turn around on this.

I got bored and it sounded fun, thanks for the interesting problem

**Review by Kylerace — CHANGES_REQUESTED** [review]

**Comment by Mooshimi** [comment]
Don't let the bot close this... it's too vital 👀

**Pull request merged**